plugin/pkg/shoot/validator/admission.go

```diff
@@ -623,10 +623,12 @@ func (c *validationContext) validateProvider(a admission.Attributes) field.Error
 
 		idxPath := path.Child("workers").Index(i)
 		if ok, validMachineTypes := validateMachineTypes(c.cloudProfile.Spec.MachineTypes, worker.Machine, oldWorker.Machine, c.cloudProfile.Spec.Regions, c.shoot.Spec.Region, worker.Zones); !ok {
-			allErrs = append(allErrs, field.NotSupported(idxPath.Child("machine", "type"), worker.Machine.Type, validMachineTypes))
+			detail := fmt.Sprintf("machine type '%s' does not support CPU architecture '%s', supported machine types of CPU architecture '%s' are: %+v", worker.Machine.Type, *worker.Machine.Architecture, *worker.Machine.Architecture, validMachineTypes)
```

Comment:
The `validateMachineTypes` function also contains a check for the machinetype in available zones, so even if the architecture matches, it might return an error, we could improve the error msg saying  some thing like "machine type does not support CPU architecture in atleast one zone"